### PR TITLE
Update demo fleet adapter integration

### PR DESCRIPTION
**Pull request by xiyuoh**
Added instructions for users to create their own fleet adapters that work with different types of robots based on the template, and a case study section for demo fleet adapter.

Note: some links will have to be updated after the demo adapter PR is merged

### CODE REVIEW

File: src/integration_fleets.md [hunk 1]
```diff
@@ -16,6 +16,9 @@ The C++ API for **Full Control** automated guided vehicle (AGV) fleets can be fo
 * [`FleetUpdateHandle`](https://github.com/open-rmf/rmf_ros2/blob/main/rmf_fleet_adapter/include/rmf_fleet_adapter/agv/FleetUpdateHandle.hpp) - Allows you to configure a fleet by adding robots and specifying settings for the fleet (e.g. specifying what types of deliveries the fleet can perform). New robots can be added to the fleet at any time.
 * [`RobotUpdateHandle`](https://github.com/open-rmf/rmf_ros2/blob/main/rmf_fleet_adapter/include/rmf_fleet_adapter/agv/RobotUpdateHandle.hpp) - Use this to update the position of a robot and to notify the adapter if the robot's progress gets interrupted.
 * [`RobotCommandHandle`](https://github.com/open-rmf/rmf_ros2/blob/main/rmf_fleet_adapter/include/rmf_fleet_adapter/agv/RobotCommandHandle.hpp) - This is a pure abstract interface class. The functions of this class must be implemented to call upon the API of the specific fleet manager that is being adapted.
+
+Note: For the **Traffic Light** category of fleet adapter (where RMF has pause/resume control over each mobile robot), the library offers an additional class:
```
Comment (Yadunund) [inline src/integration_fleets.md]: Let's rephrase this as follows and without the `Note`.

The C++ API for **Traffic Light Control** fleets (ie, fleets that only allow RMF to pause/resume each mobile robot) can also be found in the `rmf_fleet_adapter` package of the `rmf_ros2` repo. The API reuses the `Adapter` class and requires users to initialize their fleet using either of the APIs [here](https://github.com/open-rmf/rmf_ros2/blob/9b4b8a8cc38b323f875a55c70f307446584d1639/rmf_fleet_adapter/include/rmf_fleet_adapter/agv/Adapter.hpp#L106-L180). Here the user has the option to integrate via the [TrafficLight](https://github.com/open-rmf/rmf_ros2/blob/main/rmf_fleet_adapter/include/rmf_fleet_adapter/agv/TrafficLight.hpp) API or for greater convenience, via the [EasyTrafficLight](https://github.com/open-rmf/rmf_ros2/blob/9b4b8a8cc38b323f875a55c70f307446584d1639/rmf_fleet_adapter/include/rmf_fleet_adapter/agv/Adapter.hpp#L106-L180) API.

File: src/integration_fleets.md [hunk 2]
```diff
@@ -31,11 +34,53 @@ The basic workflow of developing a fleet adapter is the following:
     - You will also need to provide a callback that will be triggered when the adapter is finished registering the robot. This callback will provide you with a new `RobotUpdateHandle` for your robot. It is imperative to save this update handle so you can use it to update the robot's position over time.
 8. As new information arrives from the fleet manager API, use the collection of `RobotUpdateHandle` classes to keep the adapter up-to-date on the robots' positions.
 
+With the robots controlled and managed by their respective fleets, robots of different models with varying configurations (such as task capabilities, vehicle traits and communication interface) can coexist and operate simultaneously in the same environment under RMF.
```
Comment (Yadunund) [inline src/integration_fleets.md]: This statement seems a bit out of place. We can either move it to the `Intro` page under `What is RMF` or leave it out.

File: src/integration_fleets.md [hunk 3]
```diff
@@ -31,11 +34,53 @@ The basic workflow of developing a fleet adapter is the following:
...
 
 ## Fleet Adapter Template
-To make the process of integrating a robotic fleet with RMF even simpler, we have open-sourced a **Full Control** [template package](https://github.com/open-rmf/fleet_adapter_template) where users only need to update certain blocks of code with the API calls to their specific robot/fleet manager.
+To make the process of integrating a robotic fleet with RMF even simpler, we have open-sourced a **Full Control** [template package](https://github.com/open-rmf/fleet_adapter_template) where users only need to update certain blocks of code with the API calls to their specific robot/fleet manager. This way, users can integrate RMF while using their preferred APIs between the fleet adapter and their robots. The following diagram illustrates how RMF can communicate with fleet robots using the APIs chosen by the user and the robot vendor.
```
Comment (Yadunund) [inline src/integration_fleets.md]: Can we add a note saying that the template is just one of many ways to integrate fleets with REST/Websocket based APIs.

File: src/integration_fleets.md [hunk 4]
```diff
@@ -31,11 +34,53 @@ The basic workflow of developing a fleet adapter is the following:
...
+
+<img src="images/fleet_adapter_flow.png">
```
Comment (Yadunund) [inline src/integration_fleets.md:44]: In this diagram, let's replace the `Robot API` label with `Proprietary vendor software`

File: src/integration_fleets.md [hunk 5]
```diff
@@ -31,11 +34,53 @@ The basic workflow of developing a fleet adapter is the following:
...
+
+This fleet adapter system is also integrated in our demos worlds with simulation robots, which is further elaborated in the next section.
+
+Since the fleet adapter template already applies the C++ API and Python bindings, you can follow these steps to build your fleet adapter on top of the given template:
+1. For **each** robot fleet, create a [`config.yaml`](https://github.com/open-rmf/fleet_adapter_template/blob/main/fleet_adapter_template/config.yaml) file to include important fleet parameters. These parameters will be passed to the fleet adapter and configured when initializing the fleet.
```
Comment (Yadunund) [inline src/integration_fleets.md]: Let's mention the config at the end

File: src/integration_fleets.md [hunk 6]
```diff
@@ -31,11 +34,53 @@ The basic workflow of developing a fleet adapter is the following:
...
+
+The demos fleet adapter is integrated with the simulation robots which publish their state information via internal ROS2 messages, hence the `fleet_manager` also serves to consolidate the messages published by different robots from its fleet and sends them to the right robot's `RobotCommandHandle`. The API endpoints are designed such that the adapter can query information or send commands to a particular robot by specifying the robot's name. The `fleet_manager` will ensure that the robot exists in the fleet before returning the requested state information or relaying commands to the simulation robot. Additionally, the adapter can retrieve the status of the entire fleet's robots.
+
+### FastAPI
```
Comment (Yadunund) [inline src/integration_fleets.md]: Can leave out this portion of `FastAPI` as its an implementation detail of the demo fleet manager that we have created and not relevant to users who will look at this document to understand how to integrate their fleet with RMF. Maybe we can add this to the readme in rmf_demos_fleet_adapter instead

File: src/integration_fleets.md [hunk 7]
```diff
@@ -31,11 +34,53 @@ The basic workflow of developing a fleet adapter is the following:
...
+
+This fleet adapter implementation uses REST API with FastAPI's framework. Apart from the integrated client, users can also interact with the manager's endpoints with FastAPI's automatic documentation. This can be assessed by launching any of the demo worlds and visiting the relevant fleet's base URL followed by `/docs`. For example, to view Office world's `tinyRobot` fleet documentation, launch Office world in Gazebo with the following command:
+```
+ros2 launch rmf_demos_gz office.launch.xml
+```
+And then visit [http://127.0.0.1:22011/docs](http://127.0.0.1:22011/docs). The port number `22011` is specific to the `tinyRobot` fleet. Each fleet uses a different port number, which can be found in the respective fleet config files.
+
+### Fleet Configuration
```
Comment (Yadunund) [inline src/integration_fleets.md:68]: Same with this section. Move to readme in `rmf_demos_fleet_adapter`

**Review by Yadunund — CHANGES_REQUESTED** [review]
Thanks for this writeup! Left some comments to address.

File: src/integration_fleets.md (hunk 6 above)
Comment (xiyuoh) [reply src/integration_fleets.md]: Got it :+1: already in rmf_demos_fleet_adapter README

File: src/integration_fleets.md (hunk 5 above)
Comment (xiyuoh) [reply src/integration_fleets.md]: Done

File: src/integration_fleets.md (hunk 4 above)
Comment (xiyuoh) [reply src/integration_fleets.md:44]: Done

File: src/integration_fleets.md (hunk 3 above)
Comment (xiyuoh) [reply src/integration_fleets.md]: Done

File: src/integration_fleets.md (hunk 2 above)
Comment (xiyuoh) [reply src/integration_fleets.md]: Ok! Removed it as i think the intro explained the same idea

File: src/integration_fleets.md (hunk 1 above)
Comment (xiyuoh) [reply src/integration_fleets.md]: Done

**Review by Yadunund — APPROVED** [review]
Thanks for the changes!
